source/DataElement/AMDSFlatArray.cpp

```diff
@@ -198,3 +199,437 @@ AMDSFlatArray AMDSFlatArray::operator /(const quint32 divisor)
 
 	return newDataFlatArray;
 }
+
+bool AMDSFlatArray::setValue(int index, qint8   value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Signed8 || index < 0 || index > vectorQint8_.size()) {
+		return false;
+	}
+
+	vectorQint8_[index] = value;
+	return true;
+}
+bool AMDSFlatArray::setValue(int index, quint8  value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Unsigned8 || index < 0 || index > vectorQuint8_.size()) {
+		return false;
+	}
+
+	vectorQuint8_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, qint16  value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Signed16 || index < 0 || index > vectorQint16_.size()) {
+		return false;
+	}
+
+	vectorQint16_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, quint16 value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Unsigned16 || index < 0 || index > vectorQuint16_.size()) {
+		return false;
+	}
+
+	vectorQuint16_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, qint32  value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Signed32 || index < 0 || index > vectorQint32_.size()) {
+		return false;
+	}
+
+	vectorQint32_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, quint32 value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Unsigned32 || index < 0 || index > vectorQuint32_.size()) {
+		return false;
+	}
+
+	vectorQuint32_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, qint64  value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Signed64 || index < 0 || index > vectorQint64_.size()) {
+		return false;
+	}
+
+	vectorQint64_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, quint64 value)
+{
+	if (dataType_ != AMDSDataTypeDefinitions::Unsigned64 || index < 0 || index > vectorQuint64_.size()) {
+		return false;
+	}
+
+	vectorQuint64_[index] = value;
+	return true;
+}
+
+bool AMDSFlatArray::setValue(int index, float   value)
```

Comment:
Out of curiosity, is there a reason why this uses `float` and the next one uses `double` instead of one with `qreal`?  Since you've gone through all the trouble of using the platform independent qatomic types, it seems weird to me that qreal is not being used too.